<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1688</width>
    <height>836</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string> Sistema Gestione Studenti | Dashboard</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>.</normaloff>.</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>20</number>
    </property>
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::TabPosition::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::TabShape::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="iconSize">
       <size>
        <width>24</width>
        <height>24</height>
       </size>
      </property>
      <widget class="QWidget" name="tabRicerca">
       <attribute name="title">
        <string>Ricerca degli Studenti</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>20</number>
        </property>
        <property name="topMargin">
         <number>25</number>
        </property>
        <property name="rightMargin">
         <number>20</number>
        </property>
        <property name="bottomMargin">
         <number>20</number>
        </property>
        <item>
         <widget class="QFrame" name="frameRicerca">
          <property name="frameShape">
           <enum>QFrame::Shape::Box</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="spacing">
            <number>20</number>
           </property>
           <property name="leftMargin">
            <number>30</number>
           </property>
           <property name="topMargin">
            <number>25</number>
           </property>
           <property name="rightMargin">
            <number>30</number>
           </property>
           <property name="bottomMargin">
            <number>25</number>
           </property>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <property name="spacing">
              <number>20</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="labelMatricola">
               <property name="text">
                <string> Matricola Studente:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="lineEditCognome">
               <property name="placeholderText">
                <string>Inserisci il cognome...</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="lineEditMateria">
               <property name="placeholderText">
                <string>Inserisci il nome della materia...</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="labelMateria">
               <property name="text">
                <string>Materia di Studio:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="lineEditMatricola">
               <property name="placeholderText">
                <string>Inserisci la matricola...</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="labelCognome">
               <property name="text">
                <string> Cognome Studente:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="btnCercaPerMateria">
               <property name="toolTip">
                <string>Trova studenti iscritti a materia specifica</string>
               </property>
               <property name="text">
                <string>Ricerca Materia</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="btnCercaPerCognome">
               <property name="toolTip">
                <string>Trova studenti con cognome specifico</string>
               </property>
               <property name="text">
                <string>Ricerca Cognome</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="btnCercaPerMatricola">
               <property name="toolTip">
                <string>Trova studente per matricola specifica</string>
               </property>
               <property name="text">
                <string>Ricerca Matricola</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="btnAggiungiStudente">
             <property name="toolTip">
              <string>Registra un nuovo studente nel sistema</string>
             </property>
             <property name="text">
              <string>+Nuovo Studente</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabGestioneCorsi">
       <attribute name="title">
        <string>Gestione dei Corsi</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>25</number>
        </property>
        <property name="leftMargin">
         <number>20</number>
        </property>
        <property name="topMargin">
         <number>25</number>
        </property>
        <property name="rightMargin">
         <number>20</number>
        </property>
        <property name="bottomMargin">
         <number>20</number>
        </property>
        <item>
         <widget class="QFrame" name="frameCorsi">
          <property name="frameShape">
           <enum>QFrame::Shape::Box</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="spacing">
            <number>25</number>
           </property>
           <property name="leftMargin">
            <number>30</number>
           </property>
           <property name="topMargin">
            <number>25</number>
           </property>
           <property name="rightMargin">
            <number>30</number>
           </property>
           <property name="bottomMargin">
            <number>25</number>
           </property>
           <item>
            <layout class="QGridLayout" name="gridLayout_2">
             <property name="horizontalSpacing">
              <number>25</number>
             </property>
             <property name="verticalSpacing">
              <number>20</number>
             </property>
             <item row="0" column="0">
              <widget class="QPushButton" name="btnListaStudentiCorso">
               <property name="toolTip">
                <string>Visualizza tutti gli studenti iscritti ad un corso</string>
               </property>
               <property name="text">
                <string>Lista Studenti del Corso</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="btnDatiEsamiCorso">
               <property name="toolTip">
                <string>Mostra informazioni dettagliate sugli esami</string>
               </property>
               <property name="text">
                <string>Dati Esami del Corso</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="btnConteggioStudentiPerCorso">
               <property name="toolTip">
                <string>Statistiche numeriche studenti per ogni corso</string>
               </property>
               <property name="text">
                <string>Conteggio Studenti per Corso</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="btnConteggioMateriePerCorso">
               <property name="toolTip">
                <string>Numero di materie disponibili per corso</string>
               </property>
               <property name="text">
                <string>Conteggio Materie Corso</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frameTabella">
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="labelTitoloTabella">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="baseSize">
          <size>
           <width>1</width>
           <height>20</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 20px;</string>
         </property>
         <property name="text">
          <string>|          PARTE TABELLARE         |</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTableView" name="vistaTabella">
         <property name="editTriggers">
          <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
         </property>
         <property name="gridStyle">
          <enum>Qt::PenStyle::SolidLine</enum>
         </property>
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1688</width>
     <height>21</height>
    </rect>
   </property>
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionCaricaCSV"/>
    <addaction name="actionSalvaCSV"/>
    <addaction name="separator"/>
    <addaction name="actionEsci"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="toolTip">
    <string>Stato del sistema</string>
   </property>
  </widget>
  <action name="actionCaricaCSV">
   <property name="text">
    <string>Carica CSV (obbligatorio)</string>
   </property>
   <property name="toolTip">
    <string>Carica dati da file CSV</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSalvaCSV">
   <property name="text">
    <string>Salva file CSV</string>
   </property>
   <property name="toolTip">
    <string>Salva dati su file CSV</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionEsci">
   <property name="text">
    <string> Esci</string>
   </property>
   <property name="toolTip">
    <string>Chiudi l'applicazione</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionPulisciTabella">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="toolTip">
    <string>Svuota la tabella dei risultati</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionInfo">
   <property name="text">
    <string>Informazioni Sistema</string>
   </property>
   <property name="toolTip">
    <string>Informazioni sul sistema</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionEsci</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>699</x>
     <y>449</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
